IsotropicTextureMat.C

```diff
@@ -0,0 +1,96 @@
+// $Id$
+//==============================================================================
+//!
+//! \file IsotropicTextureMat.C
+//!
+//! \date Sep 13 2018
+//!
+//! \author Arne Morten Kvarving / SINTEF
+//!
+//! \brief Isotropic linear elastic material model defined through a texture.
+//!
+//==============================================================================
+
+#include "IsotropicTextureMat.h"
+#include "IFEM.h"
+#include "tinyxml.h"
+#include "Utilities.h"
+
+
+IsotropicTextureMat::IsotropicTextureMat (bool ps, bool ax) : 
+  LinIsotropic(ps, ax)
+{
+}
+
+
+void IsotropicTextureMat::parse (const TiXmlElement* elem)
+{
+  utl::getAttribute(elem, "file", textureFile);
+  const TiXmlElement* child = elem->FirstChildElement("range");
+  while (child) {
+    std::pair<double,double> minmax;
+    utl::getAttribute(child,"min",minmax.first); 
+    utl::getAttribute(child,"max",minmax.second);
+    
+    materials.insert(std::make_pair(minmax,LinIsotropic()));
+    materials[minmax].parse(child);
+  }
+}
+
+
+void IsotropicTextureMat::printLog () const
+{
+  for (const auto it : materials) {
+    IFEM::cout << "Material with range [" << it.first.first << "," << it.first.second << "]: ";
+    it.second.printLog();
+  }
+}
+
+
+const LinIsotropic* IsotropicTextureMat::findMaterial (const FiniteElement& fe) const
+{
+  double I = this->findIntensity(fe);
+
+  auto mat  = std::find_if(materials.begin(), materials.end(),
+            [I](const std::pair<std::pair<double,double>,LinIsotropic>& a)
+            {
+              return a.first.first >= I && a.first.second <= I;
+            });
+
+  if (mat == materials.end())
+    return nullptr;
+
+  return &mat->second;
+  
+}
+
+
+double IsotropicTextureMat::findIntensity (const FiniteElement& fe) const
+{
+  return 0.0;
+}
```

Comment:
It is not clear why you need the `findIntensity` method here. Is it supposed to be overridden by something more meaningful in a sub-class? Does not look so, since it is not declared virtual. Then why not put whatever-it-was supposed-to-do inside the `findMaterial` method.